src/platform/packages/shared/kbn-esql-ast/ADRs/ERROR_TAGGING.md

```diff
@@ -0,0 +1,140 @@
+# ADR: Error Tagging for ES|QL Validation
```

Comment:
It's an example for a small thing like this, but at least it gives you some justification and keeps the story of what we do. I think we could apply it to even more relevant things, or have a discussion about it, or ignore it for now (or forever)